### PR TITLE
[IMPROVE] Load code highlighting languages on demand and fixes on new message parser

### LINKED ISSUE #20766: Emoji in a field does not render
### Description:

Emojis used in a `field` (when using the API) do not render. Other types of markdown do seem to be working.

### Steps to reproduce:

POST https://foo.rocket.chat/api/v1/chat.postMessage
```
{
 "channel": "@username",
  "text": "Example message",
  "attachments": [
    {
      "title": "Message title",
      "text": ":smile: This works",
      "color": "#efcc96",
            "fields": [{
              "short": false,
              "title": ":frown: This does **not** work",
              "value": ":frown: This too"
            }]      
    }
  ]
}
```
### Expected behavior:

All emojis and markdown should be rendered.

### Actual behavior:

Markdown is rendered properly, but emojis in the field (both `title` and `value` sections) are not rendered. Only the raw emoji text is displayed.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.9.2
- Operating System: Linux
- NodeJS Version: 12.18.1
- MongoDB Version: 4.0.11

### Additional context

This issue occurs with both the REST API and webhooks.

Note: This issue was previously reported under #17760, but was never fixed,

### PR DESCRIPTION
**Pull request by ggazzo**
## Proposed changes (including videos or screenshots)

Now we have this setting called 'Code highlighting languages list' where you can define the languages that you want to be loaded by default.

## Issue(s)

closes #20766
closes https://github.com/RocketChat/Rocket.Chat/issues/8989
closes https://github.com/RocketChat/Rocket.Chat/issues/21184
closes https://github.com/RocketChat/Rocket.Chat/issues/7707
closes https://github.com/RocketChat/Rocket.Chat/issues/21184
closes https://github.com/RocketChat/Rocket.Chat/issues/5218
closes https://github.com/RocketChat/Rocket.Chat/issues/11675
closes https://github.com/RocketChat/Rocket.Chat/issues/19933

### CODE REVIEW

**Comment by ankar84** [comment]
probably closes https://github.com/RocketChat/Rocket.Chat/issues/22280

File: app/lib/server/startup/settings.js [hunk 1]
```diff
@@ -1286,6 +1286,12 @@ settings.addGroup('Message', function() {
 			},
 		],
 	});
+
+	this.add('Message_Code_highlight', 'javascript,css,markdown,dockerfile,json,go,rust,clean,bash,plaintext,powershell,scss,shell,yaml,vim', {
+		type: 'string',
+		public: true,
+		i18nDescription: 'Message_Code_highligh_Description',
```
Comment (sampaiodiego) [inline app/lib/server/startup/settings.js:1293]: missing i18n

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Comment by sampaiodiego** [comment]
image

maybe it is a good time do update?

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -2836,6 +2836,7 @@
   "Message_BadWordsWhitelist": "Remove words from the Blacklist",
   "Message_BadWordsWhitelistDescription": "Add a comma-separated list of words to be removed from filter",
   "Message_Characther_Limit": "Message Character Limit",
+  "Message_Code_highlight": "Message Code highlight",
```
Comment (sampaiodiego) [inline packages/rocketchat-i18n/i18n/en.i18n.json:2839]: I don't think this is explanatory enough to admins..

I'd consider something like "Code languages highlights"
And also add a description to explain better what it does, like "Comma separated list of languages (link to list of supported languages https://github.com/highlightjs/highlight.js/tree/9.18.5#supported-languages ) that will be used to highlight code blocks"

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
functionality wise it seems good.

I'll just reinforce the need of an upgrade.. current version is also showing a message on browser's console:
image

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**